12/umbraco-cms/reference/content-delivery-api/extension-points/extension-api-for-querying.md

```diff
@@ -0,0 +1,308 @@
+
+{% code title="AuthorSelector.cs" lineNumbers="true" %}
+```csharp
+using Umbraco.Cms.Core.DeliveryApi;
+using Umbraco.Cms.Core.Models;
+
+namespace Umbraco.Docs.Samples;
+
+public class AuthorSelector : ISelectorHandler, IContentIndexHandler
+{
+    private const string FeaturedAuthorsSpecifier = "featuredAuthors";
+    private const string FieldName = "featured";
+
+    // Querying
+    public bool CanHandle(string query)
+        => query.StartsWith(FeaturedAuthorsSpecifier, StringComparison.OrdinalIgnoreCase);
+
+    public SelectorOption BuildSelectorOption(string selector) =>
+        new SelectorOption
+        {
+            FieldName = FieldName,
+            Values = new[] { "y" }
+        };
+
+    // Indexing
+    public IEnumerable<IndexFieldValue> GetFieldValues(IContent content, string? culture)
+    {
+        if (content.ContentType.Alias.InvariantEquals("author") == false)
+        {
+            return Enumerable.Empty<IndexFieldValue>();
+        }
+
+        var isFeatured = content.GetValue<bool>("featured");
+
+        return new[]
+        {
+            new IndexFieldValue
+            {
+                FieldName = FieldName,
+                Values = new object[] { isFeatured ? "y" : "n" }
+            }
+        };
+    }
+
+    public IEnumerable<IndexField> GetFields()
+        => new[]
+        {
+            new IndexField
+            {
+                FieldName = FieldName,
+                FieldType = FieldType.StringRaw,
+                VariesByCulture = false
+            }
+        };
+}
+```
+{% endcode %}
+
+The `AuthorSelector` class implements the `ISelectorHandler` and `IContentIndexHandler` interfaces. The `ISelectorHandler` allows for handling the custom selector in API queries through the `CanHandle()` and `BuildSelectorOption()` methods. The implementation of `CanHandle()` determines if the given `fetch` query corresponds to the `"featuredAuthors"` value. The `BuildSelectorOption()` method constructs the selector option to search for authors with a positive value (e.g., `"y"`) in a `"featured"` index field. This implementation is very customizable, giving you the flexibility to define your own logic.&#x20;
+
+For indexing, both the `GetFieldValues()` and `GetFields()` methods play a crucial role in defining how the data should be indexed and made searchable. The `GetFieldValues()` method is responsible for retrieving the value of the _'Featured'_ field of content items of type _'author'_. Then creating an `IndexFieldValue` with the appropriate field value (`"y"` for featured, `"n"` otherwise) which will be used when populating the index. `GetFields()` defines the `"featured"` field in the **DeliveryApiContentIndex**, allowing for efficient and accurate searching.
```

Comment:
```suggestion
The `GetFields()` and `GetFieldValues()` methods each play a role in defining how the data should be indexed and made searchable.

- `GetFields() defines` the behavior of fields that are added to the index. In this example, the "featured" field is added as a "raw" string for efficient and accurate searching.
- `GetFieldValues()` is responsible for retrieving the values of the defined index fields. In this case, the 'Featured' field of content items of type 'author'. It creates an `IndexFieldValue` with the appropriate field value (`"y"` for featured, `"n"` otherwise), which will be added to the index.
```